api/client/client.go

```diff
@@ -1486,3 +1486,30 @@ func (c *Client) SetClusterAuditConfig(ctx context.Context, auditConfig types.Cl
 func (c *Client) DeleteClusterAuditConfig(ctx context.Context) error {
 	return trace.NotImplemented(notImplementedMessage)
 }
+
+// GetNetworkRestrictions retrieves the network restrictions
+func (c *Client) GetNetworkRestrictions(ctx context.Context) (*types.NetworkRestrictions, error) {
+	nr, err := c.grpc.GetNetworkRestrictions(ctx, &empty.Empty{}, c.callOpts...)
+	if err != nil {
+		return nil, trail.FromGRPC(err)
+	}
+	return nr, nil
+}
+
+// SetNetworkRestrictions updates the network restrictions
+func (c *Client) SetNetworkRestrictions(ctx context.Context, nr *types.NetworkRestrictions) error {
```

Comment:
For all setters/getters lets create an interface wrapper similar to https://github.com/gravitational/teleport/blob/master/api/types/role.go#L34 and pass it here. Let's change the API to use:

https://github.com/gravitational/teleport/blob/master/api/client/client.go#L964

This lets us to convert V3/V4 internally without changing all the methods in case of simple version updates.